packages/server-wallet/src/wallet/__test__/integration/push-message.test.ts

```diff
@@ -318,6 +318,40 @@ describe('when the application protocol returns an action', () => {
     });
   });
 
+  it.each([0, 2])(
+    'emits objectiveStarted when a %i-threaded wallet receives a new objective',
+    async () => {
+      const turnNum = 6;
+      const state = stateSignedBy()({outcome: simpleEthAllocation([]), turnNum});
+
+      const c = channel({vars: [addHash(state)]});
+      await Channel.query(wallet.knex).insert(c);
+
+      const {channelId} = c;
+      const finalState = {...state, isFinal: true, turnNum: turnNum + 1};
+
+      const callback = jest.fn();
+      wallet.once('objectiveStarted', callback);
+
+      const p = wallet.pushMessage({
+        walletVersion: WALLET_VERSION,
+        signedStates: [serializeState(stateSignedBy([bob()])(finalState))],
+        objectives: [
+          {
+            type: 'CloseChannel',
+            participants: [],
+            data: {
+              targetChannelId: channelId,
+              fundingStrategy: 'Direct',
+            },
+          },
+        ],
+      });
+      await expect(p.then(r => r.newObjectives)).resolves.toHaveLength(1);
```

Comment:
Is there a reason we don't just `await` `wallet.pushMessage` in the line above? I think it would make the assertion cleaner

Reply:
6a6a7c9da2d42b8e35d78333ec6d5fa2a9572316